### PR TITLE
[shopsys] move order and cart related logic from project-base to packages

### PR DESCRIPTION
**Pull request by vitek-rostislav**
| Q             | A
| ------------- | ---
|Description, reason for the PR| we want a clean project-base and all the logic in the packages so it is upgradeable
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

TODO:
- [x] add upgrade notes

----
:globe_with_meridians: Live Preview:
  - https://rv-order-movements.odin.shopsys.cloud
  - https://cz.rv-order-movements.odin.shopsys.cloud

### CODE REVIEW

File: packages/framework/src/Component/Packetery/PacketeryRenderer.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Packetery\Packet\PacketAttributes $packetAttributes
+     * @param \Shopsys\FrameworkBundle\Component\Packetery\PacketeryConfig $packeteryConfig
+     * @return string
+     */
+    public function getPacketXml(PacketAttributes $packetAttributes, PacketeryConfig $packeteryConfig): string
+    {
+        return $this->twig->load(self::TEMPLATE_FILE_PATH)->render([
```
Comment (grossmannmartin) [inline packages/framework/src/Component/Packetery/PacketeryRenderer.php]: I believe it should be static, maybe our ForceLateStaticBinding does not work with typed constants

File: packages/framework/src/Controller/Admin/DefaultController.php [hunk 2]
```diff
@@ -62,6 +65,8 @@ public function __construct(
      */
     public function dashboardAction(): Response
     {
+        /** @var \Shopsys\FrameworkBundle\Model\Administrator\Administrator $administrator */
+        $administrator = $this->getUser();
```
Comment (grossmannmartin) [inline packages/framework/src/Controller/Admin/DefaultController.php]: maybe we could introduce an appropriate method in AdminBaseController and do not  count on inline annotation

File: packages/framework/src/Controller/Admin/TransferIssueController.php [hunk 3]
```diff
@@ -2,46 +2,45 @@
...
+        protected readonly TransferIssueFacade $transferIssueFacade,
+        protected readonly GridFactory $gridFactory,
+        protected readonly AdministratorGridFacade $administratorGridFacade,
+        protected readonly AdministratorFacade $administratorFacade,
     ) {
     }
 
     /**
-     * @Route("/transfer/list/", name="admin_transfer_list")
+     * @Route("/transfer/issue/list/", name="admin_transferIssue_list")
```
Comment (grossmannmartin) [inline packages/framework/src/Controller/Admin/TransferIssueController.php]: Currently, we don't use camelCase in route names. I think we should not introduce a new notation style

File: packages/framework/src/Form/Constraints/UniqueFlagsValidator.php [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 declare(strict_types=1);
 
-namespace App\Form\Constraints;
+namespace Shopsys\FrameworkBundle\Form\Constraints;
```
Comment (grossmannmartin) [inline packages/framework/src/Form/Constraints/UniqueFlagsValidator.php]: I'm thinking whether such specific validators shouldn't be placed along the form type, what do you think?

File: packages/framework/src/Form/Transformers/PromoCodeFlagTransformer.php [hunk 5]
```diff
@@ -2,23 +2,24 @@
...
-namespace App\Form\Admin\Transformer;
+namespace Shopsys\FrameworkBundle\Form\Transformers;
 
-use App\Model\Order\PromoCode\PromoCodeFlag\PromoCodeFlag;
-use App\Model\Order\PromoCode\PromoCodeFlag\PromoCodeFlagFactory;
+use Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCodeFlag\PromoCodeFlag;
+use Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCodeFlag\PromoCodeFlagFactory;
 use Symfony\Component\Form\DataTransformerInterface;
 
 class PromoCodeFlagTransformer implements DataTransformerInterface
```
Comment (grossmannmartin) [inline packages/framework/src/Form/Transformers/PromoCodeFlagTransformer.php]: I would place this also along the rest of the promo code classes

File: packages/framework/src/Model/Order/PromoCode/PromoCode.php [hunk 6]
```diff
@@ -22,15 +29,51 @@ class PromoCode
 
     /**
      * @var string
-     * @ORM\Column(type="text", unique=true)
+     * @ORM\Column(type="text", unique=false)
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Order/PromoCode/PromoCode.php]: false is a default value, I believe it's not necessary to be explicit about it

File: packages/framework/src/Form/Admin/PromoCode/PromoCodeLimitType.php [hunk 7]
```diff
@@ -47,18 +47,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $options = $options['discount'];
 
         foreach ($options['constraints'] as $constraint) {
-            $constraint->groups = [PromoCodeFormTypeExtension::VALIDATION_GROUP_TYPE_PERCENT];
```
Comment (grossmannmartin) [inline packages/framework/src/Form/Admin/PromoCode/PromoCodeLimitType.php:50]: JS files for promo codes are still in project-base. Is it intentional?

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 8]
```diff
@@ -31,6 +31,9 @@ msgstr "-- Vyberte značku --"
 msgid "-- Choose unit --"
 msgstr "-- Vyberte jednotku --"
 
+msgid "-- Select name of the transfer --"
+msgstr "-- Vyber název převodu --"
```
Comment (grossmannmartin) [inline packages/framework/src/Resources/translations/messages.cs.po]: usually, we use the polite form (`vyberte`)

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 9]
```diff
@@ -2827,6 +2956,12 @@ msgstr "sazby DPH"
 msgid "VAT settings modified"
 msgstr "Nastavení DPH bylo upraveno"
 
+msgid "Valid from"
+msgstr ""
+
+msgid "Valid to"
+msgstr ""
```
Comment (grossmannmartin) [inline packages/framework/src/Resources/translations/messages.cs.po]: should this be translated too?

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 10]
```diff
@@ -2137,6 +2137,9 @@ msgstr "Export osobních údajů"
 msgid "Personal information overview"
 msgstr "Přehled osobních údajů"
 
+msgid "Personal pickup"
+msgstr ""
+
```
Comment (grossmannmartin) [inline packages/framework/src/Resources/translations/messages.cs.po]: this should be translated, right?

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 11]
```diff
@@ -2629,6 +2629,9 @@ msgstr "Některé údaje nemusí být lokalizovány nebo slouží technickým sp
 msgid "Sorting priority"
 msgstr "Priorita řazení"
 
+msgid "Standard"
+msgstr ""
```
Comment (grossmannmartin) [inline packages/framework/src/Resources/translations/messages.cs.po]: this should be translated, right?

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

File: packages/framework/src/Component/Packetery/PacketeryRenderer.php (hunk 1 above)
Comment (vitek-rostislav) [reply packages/framework/src/Component/Packetery/PacketeryRenderer.php]: fixed in 8072ac5604. What is weird, I can see quite a lot of `self` usages in the framework package, e.g. https://github.com/shopsys/shopsys/blob/3ce4b2bb94a7d95a9873ee46102a0d2e75bc87f5/packages/framework/src/Model/Seo/SeoSettingFacade.php#L38

I will create an issue for that

File: packages/framework/src/Controller/Admin/DefaultController.php (hunk 2 above)
Comment (vitek-rostislav) [reply packages/framework/src/Controller/Admin/DefaultController.php]: good idea, f01d61ae88

File: packages/framework/src/Controller/Admin/TransferIssueController.php (hunk 3 above)
Comment (vitek-rostislav) [reply packages/framework/src/Controller/Admin/TransferIssueController.php]: :+1: 55e0fbec55

File: packages/framework/src/Form/Constraints/UniqueFlagsValidator.php (hunk 4 above)
Comment (vitek-rostislav) [reply packages/framework/src/Form/Constraints/UniqueFlagsValidator.php]: yes :+1: 3638e68c80

File: packages/framework/src/Model/Order/PromoCode/PromoCode.php (hunk 6 above)
Comment (vitek-rostislav) [reply packages/framework/src/Model/Order/PromoCode/PromoCode.php]: oh, good catch, fixed in 4cab29f422

File: packages/framework/src/Form/Admin/PromoCode/PromoCodeLimitType.php (hunk 7 above)
Comment (vitek-rostislav) [reply packages/framework/src/Form/Admin/PromoCode/PromoCodeLimitType.php:50]: ah, ehm, moved along with some other things :smile: 32ec0aea24

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 8 above)
Comment (vitek-rostislav) [reply packages/framework/src/Resources/translations/messages.cs.po]: c57add06f8

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 9 above)
Comment (vitek-rostislav) [reply packages/framework/src/Resources/translations/messages.cs.po]: c57add06f8

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 10 above)
Comment (vitek-rostislav) [reply packages/framework/src/Resources/translations/messages.cs.po]: c57add06f8

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 11 above)
Comment (vitek-rostislav) [reply packages/framework/src/Resources/translations/messages.cs.po]: c57add06f8

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 9 above)
Comment (grossmannmartin) [reply packages/framework/src/Resources/translations/messages.cs.po]: Please translate also `Valid from` and `Bulk creation of promo codes` 😇

Comment (vitek-rostislav) [reply packages/framework/src/Resources/translations/messages.cs.po]: :+1: 6e366d5817

File: UPGRADE-15.0.md [hunk 12]
```diff
@@ -458,6 +458,275 @@ Follow the instructions in relevant sections, e.g. `shopsys/coding-standards` or
     -   product variants cache in `Shopsys\FrameworkBundle\Model\Product\Elasticsearch\ProductExportRepository`
 -   see #project-base-diff to update your project
 
+#### move order and cart related logic from project-base to packages ([#3088](https://github.com/shopsys/shopsys/pull/3088))
+
+-   `Shopsys\FrameworkBundle\Controller\Admin\DefaultController::__construct` interface has changed:
+    ```diff
+        public function __construct(
+            // ...
+    -       protected readonly TransferIssueFacade $transferIssueFacade,
```
Comment (grossmannmartin) [inline UPGRADE-15.0.md]: shouldn't it be added line?

**Review by grossmannmartin — APPROVED** [review]

File: UPGRADE-15.0.md (hunk 12 above)
Comment (vitek-rostislav) [reply UPGRADE-15.0.md]: yes! Thank you! :blush:

**Pull request merged**